presidio-image-redactor/tests/test_document_intelligence_ocr.py

```diff
@@ -0,0 +1,78 @@
+    #Polygon of sequence 0 are invalid
+    ({"pages":[{"words":[{"content":"Happy", "confidence": 3.14, "polygon":[]}]}]}, 
+        {"left": [0], "top": [0], "width": [0], "height": [0], "conf": [3.14], "text": ["Happy"]}),
+    #Polygon of sequence 1 are invalid
+    ({"pages":[{"words":[{"content":"Happy", "confidence": 3.14, "polygon":[{"x":1, "y":2}]}]}]}, 
+        {"left": [0], "top": [0], "width": [0], "height": [0], "conf": [3.14], "text": ["Happy"]}),
+    #Regular two point polygon
+    ({"pages":[{"words":[{"content":"Happy", "confidence": 3.14, "polygon":[{"x":1, "y":2},{"x":3, "y":42}]}]}]},
+        {"left": [1], "top": [2], "width": [2], "height": [40], "conf": [3.14], "text": ["Happy"]}),
+    #Order doesn't matter
+    ({"pages":[{"words":[{"content":"Happy", "confidence": 3.14, "polygon":[{"x":3, "y":42},{"x":1, "y":2}]}]}]},
+        {"left": [1], "top": [2], "width": [2], "height": [40], "conf": [3.14], "text": ["Happy"]}),
+    #Can specify other corners
+    ({"pages":[{"words":[{"content":"Happy", "confidence": 3.14, "polygon":[{"x":3, "y":2},{"x":1, "y":42}]}]}]},
+        {"left": [1], "top": [2], "width": [2], "height": [40], "conf": [3.14], "text": ["Happy"]}),
+],
+indirect=["ocr_response"])
+def test_given_da_response_then_get_bboxes_matches(ocr_response, expected):
+    result = DocumentIntelligenceOCR._page_to_bboxes(ocr_response.pages[0])
+    assert expected == result
+
+@pytest.mark.parametrize("ocr_response", 
+[
+    #word is incorrect
+    ({"pages":[{"word":[]}]})
+])
+def test_given_wrong_keys_in_response_then_parsing_fails_returns_exception(ocr_response):
+    with pytest.raises(AttributeError):
+        DocumentIntelligenceOCR._page_to_bboxes(ocr_response.pages[0])
+
+def test_model_id_wrong_then_raises_exception():
+    with pytest.raises(ValueError):
+        DocumentIntelligenceOCR(key="fake_key", endpoint="fake_endpoint", model_id = "fake_model_id")
+
+def test_model_id_correct_then_raises_no_exception():
+    DocumentIntelligenceOCR(key="fake_key", endpoint="fake_endpoint", model_id = "prebuilt-document")
+
+@pytest.mark.parametrize("result, ok", 
+    [
+        ({"pages": []}, False),
+        ({"pages": [{"words": []}]}, True),
+        ({"pages": [{"words": []}, {"words": []}]}, False),
+    ]
+)
+@mock.patch("presidio_image_redactor.document_intelligence_ocr.DocumentIntelligenceOCR.analyze_document")
+def test_pages_not_one_then_raises_exception(analyze_document, result, ok: bool):
+    ocr_result = AnalyzeResult.from_dict(result)
+    diOCR = DocumentIntelligenceOCR(endpoint="fake_endpoint", key="fake_key")
+    diOCR.analyze_document.return_value = ocr_result
+    if not ok:
+        with pytest.raises(ValueError):
+            diOCR.perform_ocr(b"")
+    else:
+        diOCR.perform_ocr(b"")
+
+
+#Mark this test as optionally failing
+@pytest.mark.xfail(reason="This test is expected to fail unless the environment variables are set")
```

Comment:
Instead of xfail, can we consider using `pytest.skip` or a condition inside the test checking if environment variables are available?

Reply:
Good suggestion.  I made some changes so that constructor failure results in a test skip.